src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -446,12 +446,17 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       val callKey = message.callKey
       fetchLocallyQualifiedInputs(callKey) match {
         case Success(callInputs) =>
-          val updateDbCallInputs = globalDataAccess.updateCallInputs(workflow.id, callKey, callInputs)
-          updateDbCallInputs onComplete {
-            case Success(i) =>
-              logger.debug(s"$i call input expression(s) updated in database.")
-              startActor(callKey, callInputs, message.startMode)
-            case Failure(e) => self ! AsyncFailure(new SQLException(s"Failed to update symbol inputs for ${callKey.scope.fullyQualifiedName}.${callKey.tag}.${callKey.index}", e))
+          // TODO: The block of code is only run on non-shards because of issues with DSDEEPB-2490
+          if (callKey.index.isEmpty) {
```

Comment:
Code looks good.  This should perhaps mention that this database work was unnecessary since there currently are no input symbols for shards, plus the shards appeared to be starving one another for database connections trying to do the pointless work.  i.e., this change is not skipping any database updates that we were doing before.